eve/workers/openstack-multiple-nodes/terraform/network.tf

```diff
@@ -94,3 +104,87 @@ resource "openstack_networking_secgroup_v2" "bastion" {
   name        = "${local.prefix}-bastion"
   description = "Security group for the Bastion VM"
 }
+
+resource "openstack_networking_network_v2" "control_plane" {
+  name = local.control_plane_network.name
+}
+
+resource "openstack_networking_subnet_v2" "control_plane" {
+  name        = "${local.prefix}-control-plane"
+  network_id  = openstack_networking_network_v2.control_plane.id
+  enable_dhcp = true
+  cidr        = "${local.control_plane_network.cidr}"
+  ip_version  = 4
+  gateway_ip  = "${local.control_plane_network.gateway}"
+  allocation_pool {
+    start = "${local.control_plane_network.allocation_pool_start}"
+    end   = "${local.control_plane_network.allocation_pool_end}"
+  }
+}
+
+resource "openstack_networking_secgroup_v2" "control_plane" {
+  name                 = "${local.prefix}-control-plane"
+  description          = "Security group for MetalK8s control plane"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "control_plane_ingress" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  remote_group_id   = openstack_networking_secgroup_v2.control_plane.id
+  security_group_id = openstack_networking_secgroup_v2.control_plane.id
+}
+
+resource "openstack_networking_secgroup_rule_v2" "control_plane_egress" {
+  direction         = "egress"
+  ethertype         = "IPv4"
+  remote_group_id   = openstack_networking_secgroup_v2.control_plane.id
+  security_group_id = openstack_networking_secgroup_v2.control_plane.id
+}
+
+resource "openstack_networking_port_v2" "control_plane_vip" {
+  name           = "${local.prefix}-control-plane-vip"
+  network_id     = openstack_networking_network_v2.control_plane.id
+
+  security_group_ids = [
+    openstack_networking_secgroup_v2.control_plane.id,
+  ]
+
+  fixed_ip {
+    subnet_id  = openstack_networking_subnet_v2.control_plane.id
+    ip_address = "${local.control_plane_network.vip}"
+  }
+}
+
+resource "openstack_networking_port_v2" "control_plane" {
+  name           = "${local.prefix}-control-plane-${count.index}"
+  admin_state_up = "true"
+  network_id     = openstack_networking_network_v2.control_plane.id
+
+  security_group_ids = [
+    openstack_networking_secgroup_v2.control_plane.id,
+  ]
+
+  fixed_ip {
+    subnet_id  = openstack_networking_subnet_v2.control_plane.id
+  }
+
+  allowed_address_pairs {
```

Comment:
Instead of all this mockery, if we have a private control-plane network, can't we simply get rid of port-security on it, create a port for the VIP so it's not being handed out by DHCP, and get rid of IPIP?

Reply:
We can already get rid of IPIP with the current changes, but I'll try so simplify this as you suggested.

Reply:
I tried to disable port security, but I never managed to make it work.
Every time, I end up with some (not really useful) errors when spawning instances, like this one:
```
Error: Error waiting for instance (37abba31-95c7-47df-8be5-bbf1ae8e6033) to become ready: unexpected state 'ERROR', wanted target 'ACTIVE'. last error: %!s(<nil>)
```
I'm still investigating... but I don't know if it worth spending too much time.
Still, IPIP can be removed with these changes.